File: Day344_2025/Q344_GFG.java
import java.util.*;

class Q344_GFG {
    ArrayList<Integer> findTwoElement(int arr[]) {
        int n = arr.length;

        int[] freq = new int[n];
        int rep = -1, miss = -1;

        for (int x : arr) {
            freq[x - 1]++;
        }

        for (int i = 0; i < n; i++) {
            if (freq[i] == 0) miss = i + 1;
            else if (freq[i] == 2) rep = i + 1;
        }

        ArrayList<Integer> ans = new ArrayList<>();
        ans.add(rep);
        ans.add(miss);
        return ans;
    }
}

File: Day344_2025/Q344_GFG.py
class Solution:
    def findTwoElement(self, arr):
        n = len(arr)
        freq = [0] * n
        rep = miss = -1

        for x in arr:
            freq[x - 1] += 1

        for i in range(n):
            if freq[i] == 0:
                miss = i + 1
            elif freq[i] == 2:
                rep = i + 1

        return [rep, miss]
